File: data_exports/huggingface_dataset/examples/basic_analysis.py
"""
Example: Basic analysis of Australian Health and Geographic Data

This example demonstrates how to load and analyse the AHGD dataset
using Python and common data science libraries.
"""

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

def load_dataset(format_type='parquet'):
    """Load AHGD dataset in specified format."""
    if format_type == 'parquet':
        return pd.read_parquet('ahgd_data.parquet')
    elif format_type == 'csv':
        return pd.read_csv('ahgd_data.csv')
    elif format_type == 'json':
        import json
        with open('ahgd_data.json', 'r') as f:
            data = json.load(f)
        return pd.DataFrame(data['data'])
    else:
        raise ValueError(f"Unsupported format: {format_type}")

def basic_analysis():
    """Perform basic statistical analysis."""
    # Load data
    df = load_dataset('parquet')
    
    print(f"Dataset shape: {df.shape}")
    print(f"Columns: {list(df.columns)}")
    
    # Summary statistics
    numeric_cols = df.select_dtypes(include=['float64', 'int64']).columns
    print("\nSummary Statistics:")
    print(df[numeric_cols].describe())
    
    # State-level aggregations
    if 'state_name' in df.columns and 'life_expectancy_years' in df.columns:
        state_health = df.groupby('state_name').agg({
            'life_expectancy_years': 'mean',
            'smoking_prevalence_percent': 'mean',
            'obesity_prevalence_percent': 'mean'
        }).round(2)
        
        print("\nHealth Indicators by State:")
        print(state_health)
    
    return df

def create_visualisations(df):
    """Create basic visualisations."""
    plt.style.use('seaborn-v0_8')
    
    # Life expectancy distribution
    plt.figure(figsize=(10, 6))
    plt.subplot(2, 2, 1)
    df['life_expectancy_years'].hist(bins=20, alpha=0.7)
    plt.title('Distribution of Life Expectancy')
    plt.xlabel('Years')
    
    # Health indicators correlation
    if all(col in df.columns for col in ['life_expectancy_years', 'smoking_prevalence_percent']):
        plt.subplot(2, 2, 2)
        plt.scatter(df['smoking_prevalence_percent'], df['life_expectancy_years'], alpha=0.6)
        plt.xlabel('Smoking Prevalence (%)')
        plt.ylabel('Life Expectancy (Years)')
        plt.title('Smoking vs Life Expectancy')
    
    plt.tight_layout()
    plt.savefig('ahgd_analysis.png', dpi=300, bbox_inches='tight')
    plt.show()

if __name__ == "__main__":
    # Run basic analysis
    data = basic_analysis()
    
    # Create visualisations
    create_visualisations(data)
    
    print("\nAnalysis complete! Check ahgd_analysis.png for visualisations.")

File: data_exports/huggingface_dataset/examples/basic_analysis.R
# Australian Health and Geographic Data (AHGD) - R Example
# 
# This example demonstrates how to load and analyse the AHGD dataset using R

library(arrow)      # For Parquet files
library(readr)     # For CSV files
library(jsonlite)  # For JSON files
library(dplyr)     # For data manipulation
library(ggplot2)   # For visualisation

# Load dataset (Parquet recommended for performance)
load_ahgd_data <- function(format = "parquet") {
  if (format == "parquet") {
    data <- arrow::read_parquet("ahgd_data.parquet")
  } else if (format == "csv") {
    data <- readr::read_csv("ahgd_data.csv")
  } else if (format == "json") {
    json_data <- jsonlite::fromJSON("ahgd_data.json")
    data <- as.data.frame(json_data$data)
  } else {
    stop("Unsupported format. Use 'parquet', 'csv', or 'json'")
  }
  
  return(data)
}

# Basic analysis
analyse_ahgd <- function() {
  # Load data
  df <- load_ahgd_data("parquet")
  
  cat("Dataset dimensions:", dim(df), "\n")
  cat("Column names:", paste(names(df), collapse = ", "), "\n\n")
  
  # Summary statistics for numeric columns
  numeric_cols <- sapply(df, is.numeric)
  if (any(numeric_cols)) {
    cat("Summary statistics:\n")
    print(summary(df[, numeric_cols]))
  }
  
  # State-level health indicators
  if ("state_name" %in% names(df) && "life_expectancy_years" %in% names(df)) {
    state_summary <- df %>%
      group_by(state_name) %>%
      summarise(
        avg_life_expectancy = mean(life_expectancy_years, na.rm = TRUE),
        avg_smoking = mean(smoking_prevalence_percent, na.rm = TRUE),
        avg_obesity = mean(obesity_prevalence_percent, na.rm = TRUE),
        .groups = 'drop'
      )
    
    cat("\nHealth indicators by state:\n")
    print(state_summary)
  }
  
  return(df)
}

# Create visualisations
create_plots <- function(df) {
  # Life expectancy distribution
  p1 <- ggplot(df, aes(x = life_expectancy_years)) +
    geom_histogram(bins = 20, fill = "skyblue", alpha = 0.7) +
    labs(title = "Distribution of Life Expectancy",
         x = "Life Expectancy (Years)",
         y = "Count") +
    theme_minimal()
  
  # Smoking vs Life Expectancy
  if (all(c("smoking_prevalence_percent", "life_expectancy_years") %in% names(df))) {
    p2 <- ggplot(df, aes(x = smoking_prevalence_percent, y = life_expectancy_years)) +
      geom_point(alpha = 0.6, color = "darkblue") +
      geom_smooth(method = "lm", se = TRUE, color = "red") +
      labs(title = "Smoking Prevalence vs Life Expectancy",
           x = "Smoking Prevalence (%)",
           y = "Life Expectancy (Years)") +
      theme_minimal()
    
    # Save plots
    ggsave("life_expectancy_distribution.png", p1, width = 8, height = 6, dpi = 300)
    ggsave("smoking_vs_life_expectancy.png", p2, width = 8, height = 6, dpi = 300)
  }
}

# Run analysis
main <- function() {
  cat("Loading Australian Health and Geographic Data...\n")
  data <- analyse_ahgd()
  
  cat("Creating visualisations...\n")
  create_plots(data)
  
  cat("Analysis complete!\n")
}

# Execute if run directly
if (sys.nframe() == 0) {
  main()
}
